### PR TITLE
Laboratory work #2, Budanova Tatiana - 20FPL1

### PR DESCRIPTION
**Pull request by tanynova99**

### CODE REVIEW

File: lab_2/main.py [hunk 1]
```diff
@@ -6,14 +6,29 @@
...
+
+    freqs = {}
+    full_length = len(tokens)
+    for word in tokens:
+        if word not in freqs.keys():
+            freqs[word] = 1
+        else:
+            freqs[word] += 1
+
+    freqs.update((x, round(y / full_length, ndigits=5)) for x, y in freqs.items())
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:29]: what happens here?

File: lab_2/main.py [hunk 2]
```diff
@@ -24,7 +39,21 @@ def get_language_profiles(texts_corpus: list, language_labels: list) -> dict or
...
+        return None
+    if not isinstance(language_labels, list):
+        return None
+    for label in language_labels:
+        if not isinstance(label, str):
+            return None
+
+    language_profiles = {}
+
+    for i, label in enumerate(language_labels):
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:53]: better to use names like index, etc

File: lab_2/main.py [hunk 3]
```diff
@@ -43,7 +89,25 @@ def get_text_vector(original_text: list, language_profiles: dict) -> list or Non
...
+        return None
+
+    text_vector = []
+    unique_words = get_language_features(language_profiles)
+
+    for i, word in enumerate(unique_words):
+        if word in original_text:
+            for profile in language_profiles.values():
+                if word in profile.keys():
+                    text_vector.append(profile[word])
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:106]: there is a way to get a key if it exists or 0 if it doesn't

File: lab_2/main.py (hunk 2 above)
Comment (tanynova99) [reply lab_2/main.py:57]: Okay

File: lab_2/main.py (hunk 1 above)
Comment (tanynova99) [reply lab_2/main.py:33]: I am changing the values of the keys from raw frequencies (like 5 if let's say there are 10 words) to freq / all elements resulting in asked value (like 5/10 in my example)

File: lab_2/main.py (hunk 3 above)
Comment (tanynova99) [reply lab_2/main.py:108]: For some reason it doesn't count in the last element. Please, look through what's wrong. I've been trying to find it for the past 30 mins.

File: lab_2/main.py [hunk 4]
```diff
@@ -174,7 +287,32 @@ def calculate_distance_sparse(unknown_text_vector: list,
...
+
+    if not isinstance(known_text_vector, list):
+        return None
+    for element in known_text_vector:
+        if element is None:
+            return None
+
+    distance = 0
+
+    for unknown, known in zip_longest(unknown_text_vector, known_text_vector,
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:304]: what happens here? particularly interested in `fillvalue`

File: lab_2/main.py [hunk 5]
```diff
@@ -268,15 +279,26 @@ def get_sparse_vector(original_text: list, language_profiles: dict) -> list or N
...
+    max_scores = {word: float(0) for word in unique_words}
+    # elimination of possible repeats in original text to save time
+    original_text = set(original_text)
+    # if the distance of the unique_words word is lower
+    # than the one found in language profile it's updated
+    # it's always changed from 0
+    # we get max distances of unique words
+    for profile in language_profiles.values():
+        for key, value in profile.items():
+            print(max_scores[key], value)
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:294]: better to not push prints

Comment (tanynova99) [reply lab_2/main.py:294]: Sorry I forgot to delete it ahah

**Pull request closed without merging**